packages/plugins/users-permissions/server/services/permission.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const PUBLIC_ROLE_FILTER = { role: { type: 'public' } };
+
+module.exports = ({ strapi }) => ({
+  /**
+   * Find permissions associated to a specific role ID
+   *
+   * @param {number} roleID
+   *
+   * @return {object[]}
+   */
+  async findRolePermissions(roleID) {
+    return strapi.entityService.load(
+      'plugin::users-permissions.role',
+      { id: roleID },
+      'permissions'
+    );
+  },
+
+  /**
+   * Find permissions for the public role
+   *
+   * @return {object[]}
+   */
+  async findPublicPermissions() {
+    return strapi.entityService.findMany('plugin::users-permissions.permission', {
+      where: PUBLIC_ROLE_FILTER,
+    });
+  },
+
+  /**
+   * Transform a Users-Permissions' permission into a content API one
+   *
+   * @example
+   * const upPermission = { action: 'api::foo.foo.find' };
+   *
+   * const permission = toContentAPIPermission(upPermission);
+   * // ^? { action: 'find', subject: 'api::foo.foo' }
+   *
+   * @param {object} permission
+   * @param {string} permission.action
+   *
+   * @return {{ action: string, subject: string }}
+   */
+  toContentAPIPermission(permission) {
```

Comment:
Still not sure 100% how we should treat users-permissions actions.

They're built using the controller's method name as action and the` source::api.controller` as subject (everything is then concatenated into a unique action name). 

For now I've let them as they're everywhere except when registering them into the content API permissions engine + when generating & testing abilities (in the auth strategy)

Reply:
Decided to stay consistent with how they were generated previously, but we'll need to think about it when we'll create the API token bridge with the content API engine.

**tl;dr: for now we're keeping action only**